azure-devops/azext_devops/dev/repos/policy.py

```diff
@@ -215,42 +215,104 @@ def update_policy_required_reviewer(policy_id,
+                                 allow_squash=None,
+                                 allow_rebase=None,
+                                 allow_rebase_merge=None,
+                                 allow_no_fast_forward=None,
                                  branch_match_type='exact',
                                  organization=None, project=None, detect=None):
     """Create merge strategy policy
     """
+    if (use_squash_merge is None and not allow_squash and not allow_rebase_merge and
+            not allow_rebase and not allow_no_fast_forward):
+        raise CLIError("Atleast one merge type must be enabled.")
     organization, project = resolve_instance_and_project(
         detect=detect, organization=organization, project=project)
     policy_client = get_policy_client(organization)
-    param_name_array = ['useSquashMerge']
-    param_value_array = [use_squash_merge]
+    if use_squash_merge is None:
+        param_name_array = ['allowSquash', 'allowRebase', 'allowRebaseMerge', 'allowNoFastForward']
+        param_value_array = [allow_squash, allow_rebase, allow_rebase_merge, allow_no_fast_forward]
+        for i, value in enumerate(param_value_array):
+            if value is None:
+                param_value_array[i] = False
+    else:
+        param_name_array = ['useSquashMerge']
+        param_value_array = [use_squash_merge]
+
     configuration = create_configuration_object(repository_id, branch, blocking, enabled,
                                                 'fa4e907d-c16b-4a4c-9dfa-4916e5d171ab',
                                                 param_name_array, param_value_array, branch_match_type)
-
     return policy_client.create_policy_configuration(configuration=configuration, project=project)
 
 
 def update_policy_merge_strategy(policy_id,
                                  repository_id=None, branch=None, branch_match_type=None,
                                  blocking=None, enabled=None,
                                  use_squash_merge=None,
+                                 allow_squash=None,
+                                 allow_rebase=None,
+                                 allow_rebase_merge=None,
+                                 allow_no_fast_forward=None,
                                  organization=None, project=None, detect=None):
     """Update merge strategy policy
     """
     organization, project = resolve_instance_and_project(
         detect=detect, organization=organization, project=project)
     policy_client = get_policy_client(organization)
     current_policy = policy_client.get_policy_configuration(project=project, configuration_id=policy_id)
-    param_name_array = ['useSquashMerge']
-
     current_setting = current_policy.settings
     current_scope = current_policy.settings['scope'][0]
-
-    param_value_array = [
-        use_squash_merge if use_squash_merge is not None else current_setting.get('useSquashMerge', None)
-    ]
+    current_setting_new_merge_values_array = [
```

Comment:
@gauravsaralMs @ishitam8 Please review this file again. I didn't handle update in a back compat way earlier. 
Also added tests to check deprecated option, new options and both together.